readers/api/transport.go

```diff
@@ -210,25 +218,41 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 	}
 }
 
-func authorize(r *http.Request, chanID string) error {
+func authorize(ctx context.Context, r *http.Request, chanID string) (err error) {
 	token := r.Header.Get("Authorization")
 	if token == "" {
 		return errUnauthorizedAccess
 	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-
-	_, err := auth.CanAccessByKey(ctx, &mainflux.AccessByKeyReq{Token: token, ChanID: chanID})
-	if err != nil {
-		e, ok := status.FromError(err)
-		if ok && e.Code() == codes.PermissionDenied {
-			return errUnauthorizedAccess
+	switch {
+	case strings.HasPrefix(token, thingToken):
+		token = strings.TrimPrefix(token, thingToken)
```

Comment:
The problem with this approach is that both Thing and User are `Bearers`, so the `Bearer` prefix makes sense in both cases.

Reply:
so what do you propose?
I'll remind you on this conversation https://github.com/mainflux/mainflux/pull/1470#discussion_r730734378